src/core_plugins/metric_vis/public/metric_vis_controller.js

```diff
@@ -105,6 +109,14 @@ export class MetricVisComponent extends Component {
             }
           }
 
+          let createFilter;
+          if (bucketAgg && rowHeaderIndex !== undefined) {
+            createFilter = () => {
```

Comment:
what about not creating this everytime ?
- store raw value to the `metrics`
- have one event handler on the container
- have one function instance which finds the raw value of clicked element 

its up to you